### PR TITLE
Make relayout work for `scene.camera`

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/934

so that http://codepen.io/monfera/pen/KMmPPV can now use `Plotly.relayout` as

```js
function rotate(id, angle) {
  var scene = gd._fullLayout[id];
  var camera = scene.camera;

  var rtz = xyz2rtz(camera.eye);
  rtz.t += angle;
  camera.eye = rtz2xyz(rtz);
  
  Plotly.relayout(gd, id + '.camera', camera);
}
```

instead of digging into the `scene` object internals.

gifrecord_2017-02-08_155327

----

In addition, 

- by making `relayout` work for `scene.camera` we can now rewrite the 3d camera modebar handlers in a much cleaner way https://github.com/plotly/plotly.js/commit/604f7a0a4b443c15cc6f872122150ad653305083
- to ensure that camera relayout are as fast as they can be, a new relayout path `docamera` is added in https://github.com/plotly/plotly.js/commit/99ad3cbd207e0f85319344a224c7c25f9d998f2f

### CODE REVIEW

**Comment by jackparmer** [comment]
ping @thejackluo , who was working a on remake of this NYT classic: 
https://www.nytimes.com/interactive/2015/03/19/upshot/3d-yield-curve-economic-growth.html

File: test/jasmine/tests/gl_plot_interact_test.js [hunk 1]
```diff
@@ -761,20 +763,18 @@ describe('gl2d interaction', function() {
...
-            var opts = { buttons: 1 };
-
-            mouseEvent('mousemove', start[0], start[1], opts);
-            mouseEvent('mousedown', start[0], start[1], opts);
-            mouseEvent('mousemove', end[0], end[1], opts);
-            mouseEvent('mouseup', end[0], end[1], opts);
+            mouseEvent('mousemove', start[0], start[1]);
+            mouseEvent('mousedown', start[0], start[1], { buttons: 1 });
+            mouseEvent('mousemove', end[0], end[1], { buttons: 1 });
+            mouseEvent('mouseup', end[0], end[1]);
```
Comment (alexcjohnson) [inline test/jasmine/tests/gl_plot_interact_test.js]: why is this commit in a PR about 3D?

Comment (etpinard) [inline test/jasmine/tests/gl_plot_interact_test.js]: https://github.com/plotly/plotly.js/pull/1301/commits/5f36071a2a74e0ded7400896f24d3ab22e5aa3c1 broke the test w/o me noticing. As test cases that require a gl context are skipped on CI, this went unnoticed until I now.

Comment (etpinard) [inline test/jasmine/tests/gl_plot_interact_test.js]: Full disclosure, I actually noticed this broken test case twice when releasing `1.22.0` and `1.23.0`, but I was lazy and didn't feel like making a PR _just for that_.

**Comment by alexcjohnson** [comment]
Very nice, much cleaner. 💃

**Pull request merged**